core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1860,6 +1860,30 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Submit a job for execution and return a FutureJob holding the result. Return a Future for
+   * monitoring the job success or failure event.
+   */
+  private[spark] def submitAsyncJob[T, U, R](
```

Comment:
BTW, I use the name `submitAsyncJob` because if using `submitJob`, the type inference cannot work well and many places using the previous `submitJob` cannot be compiled.